lib/app_component.dart

```diff
@@ -18,18 +19,25 @@ import 'todo_list/todo_list_component.dart';
   providers: const [materialProviders],
 )
 class AppComponent implements OnInit{
-  String date;
+  AppComponent() {
+    dateChanged.listen((val) => date = val);
+  }
+
+  @ViewChild('datepicker')
+  ElementRef inputBox;
+
+  String date = '1';
+
+  final StreamController<String> _dateChangedCtrl = new StreamController.broadcast();
+
+  Stream<String> get dateChanged => _dateChangedCtrl.stream;
 
   @override
   ngOnInit() {
-    var function = new js.JsFunction.withThis((input,val,object) {
-      date = val;
-      print(input);
-      print(val);
-      print(object);
+    var fun = new js.JsFunction.withThis((input, val, object) {
+      _dateChangedCtrl.add(val);
```

Comment:
`StreamController` and `Stream` are the equivalents to `EventEmitter` and `EventListener` in dart.

I'm not sure if this is the correct solution.